core/tee/tee_svc_storage.c

```diff
@@ -307,6 +307,12 @@ static TEE_Result tee_svc_storage_read_head(struct tee_ta_session *sess,
 
 	fops = o->pobj->fops;
 
+	o->pobj->head = malloc(sizeof(struct tee_svc_storage_head));
```

Comment:
Hmm shouldn't o->pobj->head has the latest copy (as long as there is an
open file and only malloc once)?

On Thu, Jul 14, 2016 at 11:01 PM, GuanchaoLiang <notifications@github.com>
wrote:

> In core/tee/tee_svc_storage.c
> <https://github.com/OP-TEE/optee_os/pull/912#discussion_r70925203>:
>
> > @@ -307,6 +307,12 @@ static TEE_Result tee_svc_storage_read_head(struct tee_ta_session *sess,
> >
> >  	fops = o->pobj->fops;
> >
> > +	o->pobj->head = malloc(sizeof(struct tee_svc_storage_head));
>
> It seems that, every time you want to update the head, you should read the
> head again..
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/OP-TEE/optee_os/pull/912/files/b95f28948850c1876c98d24acd925f77bbfdfde9#r70925203>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAR8h4cLIXVMCUuIoM1-NnC-FuDV_eEPks5qVyI_gaJpZM4JKg_J>
> .
>